docs/docs/unreachable.md

```diff
@@ -0,0 +1,90 @@
+the other as a major improvement. https://issues.apache.org/jira/browse/MESOS-5948. It seems as though the default Mesos behavior is as if there is no rate limit, which begs the question why we don't set DC/OS to not have this rate limit?
+
+## Marathon Unreachable Strategy
+
+Marathon has configuration options for working with unreachable tasks by setting the unreachable strategy for a Marathon application as part as the app definition:
+
+```
+unreachableStrategy: {
+"inactiveAfterSeconds": 0,
+"expungeAfterSeconds": 0
+}
+```
+
+In order for this to take effect it is necessary to get a TASK_UNREACHABLE status update from Apache Mesos. The above constraints set the minimum time of managing an unreachable strategy event. The configuration for the app definition sets up to time events. The first is `inactiveAfterSeconds` which is the time in seconds after the TASK_UNREACHABLE event from Mesos that Marathon will create a replacement task. The second configuration is `expungeAfterSeconds` which is the amount of time after the TASK_UNREACHABLE that a task will be removed killed provided 1) it was replaced with the inactiveAfterSecond event and 2) the original task (which was unreachable) becomes reachable again. From the time of `inactiveAfterSeconds` until the "expunge" event, the app will report in Marathon 2 of 1 for the number of tasks running for that app. "expunge" is a little odd, in that it is the time from TASK_UNREACHABLE. If the "inactiveAfterSeconds" trigger event trips, then the expunge will happen in the time configured regardless of if the task becomes reachable or not. In the case of a large delay in the
+return of the lost task, it will be killed immediately after it is seen by the system (assuming it is past expunge time).
+
+When the unreachable strategy is {0,0}, the replacement is near immediate with a replacement commonly running (for the test app) of roughly 3 seconds. The expunge happening around 8-11 seconds. When we include this with the Mesos details provided above for the loss of 1 agent, it means that Marathon will replace a task as soon as it is notified of the unreachable event and it will expunge the original task as soon as it is reachable. If there is time configured for the strategy, then the event cycle that Marathon uses to managing if a task should be replaced or expunged is the task reconciliation cycle. The importance of this is 2 fold: 1) the unreachable strategy time has to expire and 2) the next task reconciliation needs to happen. The task reconciliation is a Marathon system configuration [--reconciliation_interval](https://mesosphere.github.io/marathon/docs/command-line-flags.html). The Marathon defined default is 10 minutes. This means that an unreachable strategy which includes inactiveAfterSeconds = 60, will have a task replaced between 60 seconds and 10 minutes and 60 seconds. The reconciliation interval has the same effect on the expunging of a task.
```

Comment:
Where are the roughly three seconds coming from? Is there any config for that?

Reply:
I'm not sure I got this paragraph so let me rephrase it to verify.

We expunge and unreachable task once
1. A replacement became reachable
2. The `expungeAfterSeconds` expired.

Marathon only checks for the expiration on Mesos task status updates. These are usually received with each reconciliation phase. Thus it can take up to ten minutes until Marathon recognizes that `expungeAfterSeconds`.

Reply:
Let's use markdown numbered lists here, also? :-)